common/server_protocolhandler.cpp

```diff
@@ -542,6 +542,24 @@ Response::ResponseCode Server_ProtocolHandler::cmdJoinRoom(const Command_JoinRoo
     if (!r)
         return Response::RespNameNotFound;
 
+    QString roomPermission = r->getRoomPermission().toLower();
+    if (roomPermission != "none"){
+        if (roomPermission == "registered") {
```

Comment:
Switch statements in C++ do not support strings (as we described in chat).  An enum may be feasible. With the way its currently written its only called once.  The sub comparison only occurs dependent on which roomPermission level is set. 

Want to get others opinions on the enum or if the if statements should be kept.